PearlPCSup/devPearl.db

```diff
@@ -481,38 +482,78 @@ record(waveform, "$(P)$(IOC_NAME)ERROR_MSGS")
                 "MONITOR process has stopped"])
 }
 
-record(longout, "$(P)$(IOC_NAME)_ERROR_MSG_LU")
+record(longout, "$(P)_ERROR_MSG_LU")
 {
-    field(DOL, "$(P)$(IOC_NAME)ERRCODE")
+    field(DOL, "$(P)ERRCODE")
     field(OMSL, "closed_loop")
-    field(OUT, "$(P)$(IOC_NAME)_ERROR_MSG_SA.INDX PP")
+    field(OUT, "$(P)_ERROR_MSG_SA.INDX PP")
 }
 
 # INDX set by _ERROR_MSG_LU PV
-record(subArray, "$(P)$(IOC_NAME)_ERROR_MSG_SA")
+record(subArray, "$(P)_ERROR_MSG_SA")
 {
-    field(INP, "$(P)$(IOC_NAME)ERROR_MSGS")
+    field(INP, "$(P)ERROR_MSGS")
     field(FTVL, "STRING")
-    field(FLNK, "$(P)$(IOC_NAME)LAST_ERR")
+    field(FLNK, "$(P)LAST_ERR")
 }
 
-record(stringin, "$(P)$(IOC_NAME)LAST_ERR")
+record(stringin, "$(P)LAST_ERR")
 {
-    field(INP, "$(P)$(IOC_NAME)_ERROR_MSG_SA")
+    field(INP, "$(P)_ERROR_MSG_SA")
 }
 
 # log error to ioc error log
-record(stringout, "$(P)$(IOC_NAME)LAST_ERR_LOG")
+record(stringout, "$(P)LAST_ERR_LOG")
 {
-    field(DOL, "$(P)$(IOC_NAME)LAST_ERR CP")
+    field(DOL, "$(P)LAST_ERR CP")
     field(OMSL, "closed_loop")
     field(DTYP, "stdio")
     field(OUT, "@errlog")
 }
 
-record(bo, "$(P)$(IOC_NAME)ERRCODE:RESET"){
+record(bo, "$(P)ERRCODE:RESET"){
     field(DESC, "reset last error code")
     field(SCAN, "Passive")
     field(DTYP, "stream")
     field(OUT, "@PearlPC.proto reset_error $(PORT)")
 }
+
+record(ai, "$(P)PRESSURE_DIFFERENCE") {
+    field(DESC, "Pressure diff between transducers")
+	field(EGU, "bar")
+	field(HHSV, "MAJOR")
+	field(LLSV, "MAJOR")
+	field(LOLO, "-1")
+	field(HIHI, "1000") # Updated dynamically by $(P)_PRESSURE_DIFFERENCE_SET_HIHI
+	field(SCAN, "1 second")
+	field(DTYP, "stream")
+	field(INP, "@PearlPC.proto get_memory(0082) $(PORT)")
+}
+
+record(longin, "$(P)PRESSURE_DIFFERENCE_THRESHOLD") {
+    field(DESC, "Threshold for pressure difference")
+	field(SCAN, "1 second")
+	field(DTYP, "stream")
+	field(INP, "@PearlPC.proto get_memory(0081) $(PORT)")
+	field(EGU, "bar")
+}
+
+record(calcout, "$(P)_PRESSURE_DIFFERENCE_SET_HIHI") {
+    field(INPA, "$(P)PRESSURE_DIFFERENCE_THRESHOLD CP")
+	field(CALC, "A")
+    field(OUT, "$(P)PRESSURE_DIFFERENCE.HIHI")
+}
+
+record(longout, "$(P)PRESSURE_DIFFERENCE_THRESHOLD:SP") {
+    field(DESC, "Setpoint threshold for pressure diff.")
+	field(EGU, "bar")
+	# Sent to hardware by $(P)SEND_PARAMETERS
+}
+
+record(longout, "$(P)PRESSURE_DIFFERENCE_THRESHOLD:SP:OUT"){
```

Comment:
Could we maybe shorten some of this PV name a little - IN:PEARL:PEARLPC_01:PRESSURE_DIFFERENCE_THRESHOLD:SP:OUT is 56 characters but doesn't give a lot of scope for e.g. longer instrument names        PRESSURE_DIFF_THOLD  maybe?